nailgun/entities.py

```diff
@@ -3330,13 +3368,19 @@ def update_payload(self, fields=None):
         return {u'setting': super(Setting, self).update_payload(fields)}
 
 
-class SmartProxy(Entity, EntityReadMixin, EntitySearchMixin):
+class SmartProxy(
+        Entity,
+        EntityReadMixin,
+        EntitySearchMixin,
+        EntityUpdateMixin):
     """A representation of a Smart Proxy entity."""
 
     def __init__(self, server_config=None, **kwargs):
         self._fields = {
             'name': entity_fields.StringField(required=True),
             'url': entity_fields.URLField(required=True),
+            'location': entity_fields.OneToManyField(Location),
+            'organization': entity_fields.OneToManyField(Organization),
```

Comment:
Good choice. These should both be `OneToManyField`s.

This is a little bit confusing, because `GET /api/smart_proxies` only lets you search by one location or organization at a time:

    GET /api/smart_proxies
        List all smart proxies
        location_id: Scope by locations
        organization_id: Scope by organizations

However, the POST and PUT calls let you work with multiple locations or organizations. For example:

    POST /api/smart_proxies
        smart_proxy[location_ids]: REPLACE locations with given ids
        smart_proxy[organization_ids]: REPLACE organizations with given ids
        ...

The take-away is that the `SmartProxy` entity really does have one-to-many relationships with organizations and locations, and the GET call is just limited.